xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/PVR.h

```diff
@@ -0,0 +1,3076 @@
+  }
+
+  inline static const char* ADDON_GetBackendName(const AddonInstance_PVR* instance)
+  {
+    static_cast<CInstancePVRClient*>(instance->toAddon->addonInstance)->m_backendName =
+        static_cast<CInstancePVRClient*>(instance->toAddon->addonInstance)->GetBackendName();
+    return static_cast<CInstancePVRClient*>(instance->toAddon->addonInstance)
+        ->m_backendName.c_str();
+  }
+
+  inline static const char* ADDON_GetBackendVersion(const AddonInstance_PVR* instance)
+  {
+    static_cast<CInstancePVRClient*>(instance->toAddon->addonInstance)->m_backendVersion =
+        static_cast<CInstancePVRClient*>(instance->toAddon->addonInstance)->GetBackendVersion();
+    return static_cast<CInstancePVRClient*>(instance->toAddon->addonInstance)
+        ->m_backendVersion.c_str();
+  }
+
+  inline static const char* ADDON_GetBackendHostname(const AddonInstance_PVR* instance)
+  {
+    static_cast<CInstancePVRClient*>(instance->toAddon->addonInstance)->m_backendHostname =
+        static_cast<CInstancePVRClient*>(instance->toAddon->addonInstance)->GetBackendHostname();
+    return static_cast<CInstancePVRClient*>(instance->toAddon->addonInstance)
+        ->m_backendHostname.c_str();
+  }
+
+  inline static const char* ADDON_GetConnectionString(const AddonInstance_PVR* instance)
+  {
+    static_cast<CInstancePVRClient*>(instance->toAddon->addonInstance)->m_connectionString =
+        static_cast<CInstancePVRClient*>(instance->toAddon->addonInstance)->GetConnectionString();
+    return static_cast<CInstancePVRClient*>(instance->toAddon->addonInstance)
+        ->m_connectionString.c_str();
+  }
+
+  inline static PVR_ERROR ADDON_GetDriveSpace(const AddonInstance_PVR* instance,
+                                              uint64_t* total,
+                                              uint64_t* used)
+  {
+    return static_cast<CInstancePVRClient*>(instance->toAddon->addonInstance)
+        ->GetDriveSpace(*total, *used);
+  }
+
+  inline static PVR_ERROR ADDON_CallSettingsMenuHook(const AddonInstance_PVR* instance,
+                                                     const PVR_MENUHOOK* menuhook)
+  {
+    return static_cast<CInstancePVRClient*>(instance->toAddon->addonInstance)
+        ->CallSettingsMenuHook(menuhook);
+  }
+
+  //--==----==----==----==----==----==----==----==----==----==----==----==----==
+
+  inline static int ADDON_GetChannelsAmount(const AddonInstance_PVR* instance)
+  {
+    return static_cast<CInstancePVRClient*>(instance->toAddon->addonInstance)->GetChannelsAmount();
+  }
+
+  inline static PVR_ERROR ADDON_GetChannels(const AddonInstance_PVR* instance,
+                                            ADDON_HANDLE handle,
+                                            bool radio)
+  {
+    std::vector<PVRChannel> channels;
+    PVR_ERROR error = static_cast<CInstancePVRClient*>(instance->toAddon->addonInstance)
+                          ->GetChannels(radio, channels);
+    for (auto iter : channels)
```

Comment:
in case of error, I would not call the callback.

Reply:
Will add a check here and on other places, also change to "const auto&"